Formula/emacs-plus@31.rb

```diff
@@ -66,6 +67,12 @@ class EmacsPlusAT31 < EmacsBase
     depends_on "zlib" => :build
   end
 
+  if build.with? "mps"
+    depends_on "libmps" => :recommended
+    # Set the HOMEBREW_EMACS_PLUS_31_BRANCH environment variable
+    ENV["HOMEBREW_EMACS_PLUS_31_BRANCH"] = ENV.fetch("HOMEBREW_EMACS_PLUS_31_BRANCH", "scratch/igc")
```

Comment:
No need to modify ENV for this. You can simply use different branches based on build options below.